test/server/cards/06_SEC/upgrades/NimbleProwess.spec.ts

```diff
@@ -0,0 +1,49 @@
+describe('Nimble Prowess', function () {
+    integration(function (contextRef) {
+        describe('Nimble Prowess\'s ability', function () {
+            beforeEach(async function () {
+                await contextRef.setupTestAsync({
+                    phase: 'action',
+                    player1: {
+                        hand: ['nimble-prowess'],
+                        groundArena: ['battlefield-marine'],
+                        spaceArena: ['awing']
+                    },
+                    player2: {
+                        groundArena: ['wampa', 'atst'],
+                        spaceArena: ['green-squadron-awing', 'phoenix-squadron-awing']
+                    }
+                });
+            });
```

Comment:
We've had issues in the past with upgrades that have "friendly unit" restrictions combined with either taking control and moving the upgrade or playing from opponent's discard. Could you add a test for one of those scenarios just for completeness sake? I'd say just one test with either A FIne Addition or Hondo Ohnaka should work.